modules/live-schedule-impl/src/main/java/org/opencastproject/liveschedule/impl/LiveScheduleServiceImpl.java

```diff
@@ -240,14 +255,23 @@ public boolean deleteLiveEvent(String mpId) throws LiveScheduleException {
     if (mp == null) {
       logger.debug("Live media package {} not found in search index", mpId);
       return false;
-    } else
+    } else {
+      if (!isLive(mp)) {
+        logger.info("Media package {} is not live. Not retracting.", mpId);
+        return false;
+      }
       return retractLiveEvent(mp);
```

Comment:
No strong feelings about this and it depends if you really want/need logging here but the whole this is just:

```java
return mp != null
    && isLive(mp)
    && retractLiveEvent(mp);
```

…which might be easier to read/understand. I'll leave it to you to decide what you want to use ;-P
A similar handling would be possible in the other methods btw.

Reply:
I agree that it would be more compact but, since you said you don't have strong feelings, I prefer to keep the logging. It has proved to be very helpful when debugging problems.

Reply:
ok